### ISSUE TITLE
Add Rx Support

### ISSUE THREAD

**Issue by drmaas** [AsyncHttpClient/async-http-client#953]
Add optional Rx support to the AsyncHttpClient, e.g.
```
...
AsyncHttpClient asyncHttpClient = new AsyncHttpClient();
Observable<Response> observable = asyncHttpClient.prepareGet("http://www.ning.com/").observe();
observable.subscribe((response) -> {
  System.out.println(response.getResponseBody());
});
...
```
The RxJava dependency could be optional, such that consumers would provide it only if they desire to use it. 

Another option would be to create a new client class that could inherit most behavior from the existing client, but provide the additional Rx support.

**Comment by slandelle** [comment]
Is this something you plan on contributing?

**Comment by drmaas** [comment]
I certainly can start on it. I have a simple version running locally. I will need to flesh out the design and add tests.

**Comment by slandelle** [comment]
Yep. We need a new adapter module so that core API doesn't depend on Rx.

**Comment by slandelle** [comment]
Closed by #961
